src/App.tsx

```diff
@@ -15,49 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function App() {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const sortAlphabetically = () => {
+    setIsReversed(false);
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setIsReversed(false);
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setIsReversed((prevState) => !prevState);
+  };
+
+  const getReorderedGoods = (): string[] => {
+    let visibleGoods = [...goodsFromServer];
+
+    visibleGoods.sort((a, b) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods = visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
```

Comment:
Why did this function still remain in `App` component ?
Remove it from app component